source/fundamentals/aggregation-expression-operations.txt

```diff
@@ -0,0 +1,1221 @@
+company from the employees' last names and employee IDs.
+
+The ``append()`` method combines the ``firstName`` and ``lastName`` fields into
+a single username, while the ``toLower()`` method makes the entire username
+lowercase. This example stores the result as a new ``username`` field using
+the ``project()`` method.
+
+The following code shows the pipeline for this aggregation:
+
+.. code-block:: java
+   :copyable: true
+
+   var lastName = current().getString("lastName");
+   var employeeID = current().getString("employeeID");
+
+   asList(project(fields(
+       computed("username", lastName
+           .append(employeeID)
+           .toLower())
+   )));
+   
+The following code provides an equivalent aggregation pipeline in
+the Query API:
+
+.. code-block:: javascript
+   :copyable: true
+
+   [ { $project: {
+       username: {
+           $toLower: { $concat: ["$lastName", "$employeeID"] } }
+   } } ]
+
+Type-Checking Operations
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can perform a type-check operation on a value of type ``MqlValue``
+using the methods described in this section.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 50 50
+
+   * - Java Method
+     - Aggregation Pipeline Operator
+
+   * - `isArrayOr() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#isArrayOr(com.mongodb.client.model.mql.MqlArray)>`__
+     - *No corresponding operator*
+
+   * - `isBooleanOr() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#isBooleanOr(com.mongodb.client.model.mql.MqlBoolean)>`__
+     - *No corresponding operator*
+
+   * - `isDateOr() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#isDateOr(com.mongodb.client.model.mql.MqlDate)>`__
+     - *No corresponding operator*
+
+   * - `isDocumentOr() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#isDocumentOr(T)>`__
+     - *No corresponding operator*
+
+   * - `isIntegerOr() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#isIntegerOr(com.mongodb.client.model.mql.MqlInteger)>`__
+     - *No corresponding operator*
+
+   * - `isMapOr() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#isMapOr(com.mongodb.client.model.mql.MqlMap)>`__
+     - *No corresponding operator*
+
+   * - `isNumberOr() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#isNumberOr(com.mongodb.client.model.mql.MqlNumber)>`__
+     - *No corresponding operator*
+
+   * - `isStringOr() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#isStringOr(com.mongodb.client.model.mql.MqlString)>`__
+     - *No corresponding operator*
+
+Suppose you have a collection of rating data. An early version of the review
+schema allowed users to submit reviews without a star rating. You want to clean up
+the data, and convert any reviews without a star rating to be the minimum
+value of 1 star.
```

Comment:
...but what if the review was very good!

Just a note - this is entirely optional - it seems a bit more usual to apply a general rating to a missing specific rating:

```
var comfort = current().getField("comfort");
var overall = current().getField("overall");

computed("comfort", comfort.isNumberOr(overall))
```

Reply:
Fair point - I was trying to make it be a case where reviews used to be 0-based for negative reviews and the system is going to 1-based for negative reviews. I added some clarification to specify this was just for negative reviews.